lib/fast_versioning/timelines/value_durations.rb

```diff
@@ -0,0 +1,58 @@
+module FastVersioning
+  # represents collection of durations
+  module Timelines
+    class ValueDurations
+      # @param durations [Array(Duration)]
+      # @param value [String, nil] value to filter by, nil = all
+      def initialize(durations:, value: nil)
+        self.durations = durations
+        self.value = value
+      end
+
+      # @return [Array] unique values for filtered duration items
+      def uniq_values
+        filtered_durations.map(&:value).uniq
+      end
+
+      # @return [Array]
+      def date_ranges
+        filtered_durations.map(&:date_range)
+      end
+
+      # @return [Hash] a hash of value => duration date ranges
+      # @example
+      #   {
+      #     "active" => [
+      #         Sat, 03 Apr 2021 00:00:00 EDT -04:00..Mon, 05 Apr 2021 21:53:48 EDT -04:00,
+      #         Mon, 05 Apr 2021 22:02:44 EDT -04:00..Wed, 07 Apr 2021 08:41:55 EDT -04:00
+      #     ],
+      #     "inactive" => [
+      #         Mon, 05 Apr 2021 21:53:48 EDT -04:00..Mon, 05 Apr 2021 22:02:44 EDT -04:00
+      #     ]
+      #   }
+      def to_h
+        uniq_values.map do |value|
+          [
+            value,
+            ValueDurations.new(durations: filtered_durations, value: value).date_ranges.compact
```

Comment:
when would compact come into play? Would that happen if you passed a `value` that wasn't in the fast versions for a particular item?

Reply:
ah, good catch - that was the `starting_from` optional limit that could cause [nil values to be returned](https://github.com/ArcadiaPower/fast-versioning/pull/14/commits/a24fd43a7719166d987cbab8eddf07b1023248e0#diff-94e6f22e72f4b22815905c0234e5b824220d4ede7870827eaaf438bdf118ff75R17) but I removed this altogether so that's no longer needed, I'll remove it, thanks!